plugins/MauticFullContactBundle/Controller/PublicController.php

```diff
@@ -0,0 +1,313 @@
+                $socialProfiles = $result['socialProfiles'];
+            }
+            foreach (['facebook', 'foursquare', 'googleplus', 'instagram', 'linkedin', 'twitter'] as $p) {
+                foreach ($socialProfiles as $socialProfile) {
+                    if (array_key_exists('type', $socialProfile) && $socialProfile['type'] === $p) {
+                        $social[$p] = array_key_exists('url', $socialProfile) ? $socialProfile['url'] : '';
+                        break;
+                    }
+                }
+            }
+
+            $data = [];
+
+            if (array_key_exists('contactInfo', $result)) {
+                $data = [
+                    'lastname' => array_key_exists(
+                        'familyName',
+                        $result['contactInfo']
+                    ) ? $result['contactInfo']['familyName'] : '',
+                    'firstname' => array_key_exists(
+                        'givenName',
+                        $result['contactInfo']
+                    ) ? $result['contactInfo']['givenName'] : '',
+                    'website' => (array_key_exists('websites', $result['contactInfo']) && count(
+                            $result['contactInfo']['websites']
+                        )) ? $result['contactInfo']['websites'][0]['url'] : '',
+                    'skype' => (array_key_exists('chats', $result['contactInfo']) && array_key_exists(
+                            'skype',
+                            $result['contactInfo']['chats']
+                        )) ? $result['contactInfo']['chats']['skype']['handle'] : '',
+                ];
+            }
+            $data = array_merge(
+                $data,
+                [
+                    'company'  => (null !== $org) ? $org['name'] : '',
+                    'position' => (null !== $org) ? $org['title'] : '',
+                    'city'     => (null !== $loc && array_key_exists('city', $loc) && array_key_exists(
+                            'name',
+                            $loc['city']
+                        )) ? $loc['city']['name'] : '',
+                    'state' => (null !== $loc && array_key_exists('state', $loc) && array_key_exists(
+                            'name',
+                            $loc['state']
+                        )) ? $loc['state']['name'] : '',
+                    'country' => (null !== $loc && array_key_exists('country', $loc) && array_key_exists(
+                            'name',
+                            $loc['country']
+                        )) ? $loc['country']['name'] : '',
+                ]
+            );
+
+            $data = array_merge($data, $social);
+
+            /** @var \Mautic\LeadBundle\Model\LeadModel $model */
+            $model = $this->getModel('lead');
+            /** @var Lead $lead */
+            $lead = $model->getEntity($id);
+            $model->setFieldValues($lead, $data);
+            $model->saveEntity($lead);
```

Comment:
Need to prevent a loop here with auto updating with the service. This will call your LeadSubscriber event that executes the lookup that calls the callback that calls the LeadSubscriber listener that executes the lookup... We want to use this method so that listeners will know when a contact's details are update by the callback but we want to full contact plugin's listener to not re-dispatch. Can probably just set a public property on $lead that's checked for in LeadSubscriber.php